### ISSUE TITLE
Not compiling on emacs-26?

### ISSUE THREAD

**Issue by jwiegley** [larsbrinkhoff/emacs-cl#7]
I get this when running `make` in `src`:

```
emacs -batch -l load-cl.el -f compile-cl
Loading /Users/johnw/emacs/site-lisp/emacs-cl/src/utils.el (source)...
Loading /Users/johnw/emacs/site-lisp/emacs-cl/src/func.el (source)...
Loading /Users/johnw/emacs/site-lisp/emacs-cl/src/cl-evaluation.el (source)...
Loading /Users/johnw/emacs/site-lisp/emacs-cl/src/cl-flow.el (source)...
Loading /Users/johnw/emacs/site-lisp/emacs-cl/src/cl-numbers.el (source)...
Eager macro-expansion failure: (error "Autoloading file /nix/store/g5fgi3ivqd05habgdkhh8nxykxyx88gn-emacs26/share/emacs/26.0.90/lisp/emacs-lisp/cl-extra.elc failed to define function cl-mapcar-many")
Debugger entered--Lisp error: (error "Autoloading file /nix/store/g5fgi3ivqd05habgdkhh8nxykxyx88gn-emacs26/share/emacs/26.0.90/lisp/emacs-lisp/cl-extra.elc failed to define function cl-mapcar-many")
  (cl-mapcar-many fn lists)
  (if (null (cdr lists)) (mapcar fn (car lists)) (cl-mapcar-many fn lists))
  (apply (function append) (if (null (cdr lists)) (mapcar fn (car lists)) (cl-mapcar-many fn lists)))
  mappend((lambda (var default) (list (list 'when (list 'eq var (list 'quote unbound)) (list 'setq var (if env (compile-form default env) default))))) (START END RADIX JUNK-ALLOWED) (0 (LENGTH string) 10 nil))
```

**Comment by larsbrinkhoff** [comment]
Thanks, I'll look into it.  No, I haven't tested it with the latest Emacs versions.

**Comment by larsbrinkhoff** [comment]
The primary cause seems to be eager macro expansion, which didn't exist when I wrote this ages ago.  I'm trying to rearrange code to resolve dependencies betwen macros and functions.

**Comment by larsbrinkhoff** [comment]
I'd like to fix this, but I have been busy with other tasks.  @jwiegley, I'm guessing you don't have an urgent need for this to work?

**Comment by jwiegley** [comment]
No, none at all. If the answer is "check back in 2020", I'm fine with that. :)

**Comment by priyadarshan** [comment]
> "check back in 2020"

I also would be quite interested in using `emacs-cl` on Emacs 27.1.

Is there a chance for this to be fixed?

**Comment by larsbrinkhoff** [comment]
Thanks for the timely comment!  I do want emacs-cl to work with the latest Emacs releases.  I see a potential in the future for Emacs libraries to be able to use Common Lisp libraries, and I would be excited to see where that might lead.  But unfortunatly, I'm more busy than ever, so I don't have time to work on this now.

Check back in 2030? ;-)

That said, I would be happy to work together with some contributors who are interested in resolving this.  No code in emacs-cl is sacred, so I'm willing to see drastic changes and throw out support for older Emacs versions if need be.

**Comment by aindilis** [comment]
I don't have an urgent or serious need, nor the skill to help, but I am interested in seeing if I couldn't use Emacs-CL for improving Emacs interaction with CYC through SubL.

**Comment by larsbrinkhoff** [comment]
Hello @aindilis,
Thanks for your interest.  I still would like to see emacs-cl move forward, but I'm busy doing other things.

Two things stand out in particular:
- Use the new compiler/byte code support for lexical variables.
- Use the new class instance support for CLOS.
